src/features/foods/components/Form/EditDrawerContent/index.tsx

```diff
@@ -36,13 +42,42 @@ export const EditDrawerContent = ({
   containerIdNameMap,
   groupIdNameMap,
 }: IEditDrawerContentProps) => {
+  /**
+   * Process when the delete button is clicked
+   */
   const handleDeleteClick = () => {
     alert('Successfully deleted!');
     setIsDrawerOpen(false);
   };
 
-  const handleUpdateClick = () => {
-    alert('Successfully updated!');
+  const form = useForm<FoodInputs>({
+    resolver: zodResolver(foodFormSchema),
+  });
+
+  /**
+   * 1. Initialize the form values when the drawer gets opened.
+   * 2. Clear the form values when the drawer gets closed, since the state variable `food` becomes undefined.
+   */
+  useEffect(() => {
```

Comment:
I just wondered if the nullish coalescing operator in line 64 and 66 are necessary...
Didn't cause error without them, like this ⬇️  
```tsx
useEffect(() => {
    form.reset({
      name: food?.name ?? '',
      group: containerIdGroupIdMap[food?.containerId ?? ''],
      container: food?.containerId ?? '',
      quantity: String(food?.quantity ?? ''),
      unit: food?.unit ?? '',
      expiry: food?.expiry,
      category: food?.category ?? 'unselected',
    });
  }, [containerIdGroupIdMap, food, form]);
```
It's really minor thing tho! Please tell me if it's necessary, I can learn from that!

Reply:
@kanta1207 
Yes, I think it's necessary.

It causes an error if `null` and `undefined` are not handled appropriately, which says like:

> A component is changing an uncontrolled input to be controlled

See: https://react.dev/reference/react-dom/components/textarea#im-getting-an-error-a-component-is-changing-an-uncontrolled-input-to-be-controlled

Reply:
The reason why you don't see errors may be because all the properties of the IFood' are currently required. I you change. You would see some error by tsc if you chage

```ts
export interface IFood {
  id: string;
  name: string;
  quantity: number;
  category: string;
  unit: string | null;
  expiry: Date | null;
  createdAt: Date;
}
```
to

```ts
export interface IFood {
  id: string;
  name: string;
  quantity: number;
  category: string;
  unit: string;
  expiry: Date;
  createdAt: Date;
}
```

Reply:
I am already addressing this issue, and as soon as this PR is merged, I will create another PR for this.

https://github.com/nishiki-tech/nishiki-frontend/issues/220

Reply:
Changing it to this ⬇️ , right? unit and expiry are not nullable originally.
```tsx
export interface IFood {
  id: string;
  name: string;
  quantity: number;
  category: string;
  unit: string | null;
  expiry: Date | null;
  createdAt: Date;
}
```
Indeed, changing some properties nullable caused type errors(Not sure if I'm producing the same error tho).
I honestly couldn't understand the necessarity 100%, but let me deepen my understanding more on your next PR, since it's probably not a huge deal and I'm sure you're handling things with certain idea!